### PR TITLE
test: Updated asserts in test_io

### PR DESCRIPTION
**Pull request by frgfm**
This PR updates all raw `assert` to the corresponding `unittest.TestCase` assert method for tests related to `torchvision.io` as suggested by #1483.
The specific test was run locally with success, any feedback is welcome!

### CODE REVIEW

File: test/test_io.py [hunk 1]
```diff
@@ -53,8 +53,8 @@ def _create_video_frames(num_frames, height, width):
 @contextlib.contextmanager
 def temp_video(num_frames, height, width, fps, lossless=False, video_codec=None, options=None):
     if lossless:
-        assert video_codec is None, "video_codec can't be specified together with lossless"
-        assert options is None, "options can't be specified together with lossless"
+        self.assertIsInstance(video_codec, type(None), "video_codec can't be specified together with lossless")
```
Comment (fmassa) [inline test/test_io.py]: This is not correct, `temp_video` is not a class method but a free function, so it doesn't have a `self` method.

Comment (frgfm) [reply test/test_io.py]: My bad, I had the same issue in the first place but didn't commit the right version of it

File: test/test_io.py [hunk 2]
```diff
@@ -51,10 +51,10 @@ def _create_video_frames(num_frames, height, width):
 
 
 @contextlib.contextmanager
-def temp_video(num_frames, height, width, fps, lossless=False, video_codec=None, options=None):
+def temp_video(tester, num_frames, height, width, fps, lossless=False, video_codec=None, options=None):
```
Comment (fmassa) [inline test/test_io.py]: I don't think this is a good idea. This is a function that is independent on the `tester`, so I think it should stay free from it.

From my view, if you want to remove the `assert` here  (which is outside of the test btw), I'd do something like
```python
if video_codec is not None:
    raise ValueError(...)
```

Comment (frgfm) [reply test/test_io.py]: Good idea, I'll update the PR. Two quick questions:
- would you prefer a `AssertionErrorr` or `ValueError`?
- was it for error clarity that both verifications are not performed at once?

**Pull request merged**